src/dpapimk_fmt_plug.c

```diff
@@ -182,11 +183,11 @@ static int valid(char *ciphertext, struct fmt_main *self)
 		goto err;
 	if (!isdec(p))
 		goto err;
-	if (!atoi(p))
-		goto err;
+	length1 = atoi(p);
```

Comment:
Not specific to this format, but in general the string containing only decimal digits isn't enough to make sure `atoi()` is formally safe to use. Signed integer overflow is UB, and too many decimal digits can trigger just that. We should avoid `atoi()` altogether.

Reply:
I just looked at `isdec*()` in `common.c`, and they're totally awful. :-( On invalid input, they would cause overflow, then try to post-check for it, but the overflow itself is already UB. Those functions should be rewritten to manually work on the strings of digits. If we introduce a limit of at most 9 digits in there, then it would in fact be safe for us to call `atoi()` next.

Reply:
https://github.com/magnumripper/JohnTheRipper/issues/2951 is relevant here.